sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/SignInResult.java

```diff
@@ -0,0 +1,46 @@
+package com.microsoft.appcenter.identity;
+
+/**
+ * Result for sign-in operation.
+ */
+public class SignInResult {
+
+    /**
+     * User information.
+     */
+    private final UserInformation mUserInformation;
+
+    /**
+     * Error that may have occurred during sign-in.
+     */
+    private final Exception mException;
+
+    /**
+     * Init.
+     *
+     * @param userInformation User information.
+     * @param exception       Exception if an error occurred during sign-in.
+     */
+    SignInResult(UserInformation userInformation, Exception exception) {
+        mUserInformation = userInformation;
+        mException = exception;
+    }
+
+    /**
+     * Get error if any occurred.
```

Comment:
We are returning an exception. *Get an exception of sign-in if it failed* maybe?